contracts/payment/SplitPayment.sol

```diff
@@ -32,7 +33,11 @@ contract SplitPayment {
   /**
    * @dev payable fallback
    */
-  function () external payable {}
+  function () external payable {
+    uint256 newFunds = msg.value;
+    uint256 fundsPerShare = newFunds.div(totalShares);
```

Comment:
Isn't there an issue where with integer division? The current implementation multiplies by `shares[payee]` first, helping with precision.

Though I guess Ether units are so small anyway that this is a non-issue.

Reply:
@nventuro Thanks for looking this over. To be honest I am unsure how to test precision and best practices for integer division in smart contracts. For example, what happens when `msg.value` is very small? I don't think these mods change that scenario.

Also, if variables like `totalReleased` are unused, should they generally be removed?

Reply:
No worries, I'm also unsure regarding this. Multiplying first is definitely better, but even the current implementation has issues if a party holds only a small percentage of shares.

Regarding unused state variables: yes, I'd remove them, since they cause extra gas costs for no benefit (and such an accounting task could be done off-chain anyway).

Reply:
> even the current implementation has issues if a party holds only a small percentage of shares.

Not really. That shareholder will be able to claim as much ether as corresponds to them, but naturally only in multiples of 1 wei, which is the smallest unit. Amounts less than 1 wei are kept in the contract but they are still the shareholder's, and can be claimed once enough ether accumulates again.

This is something that the proposed implementation doesn't handle well.